pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -221,31 +222,49 @@ func (p *Plugin) Attest(ctx context.Context, req *workloadattestorv1.AttestReque
 
 	// Poll pod information and search for the pod with the container. If
 	// the pod is not found then delay for a little bit and try again.
+	var scratch []byte
 	for attempt := 1; ; attempt++ {
 		log = log.With(telemetry.Attempt, attempt)
 
-		list, err := config.Client.GetPodList()
+		podListBytes, err := config.Client.GetPodList()
 		if err != nil {
 			return nil, err
 		}
 
+		var parser fastjson.Parser
+		podList, err := parser.ParseBytes(podListBytes)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "unable to parse kubelet response: %v", err)
+		}
+
 		var attestResponse *workloadattestorv1.AttestResponse
-		for _, item := range list.Items {
-			item := item
-			if podKnown && item.UID != podUID {
-				// The pod holding the container is known. Skip unrelated pods.
-				continue
+		for _, podValue := range podList.GetArray("items") {
+			if podKnown {
+				uidBytes := podValue.Get("metadata", "uid").GetStringBytes()
```

Comment:
I waffled on that but ultimately decided that the way this is written preserves existing behavior since we weren't checking for, and logging, a corev1.Pod struct that had an empty UID previously.